lib/Driver/Compilation.cpp

```diff
@@ -1645,6 +1680,13 @@ namespace driver {
           .findExternallyDependentUntracedJobs(externalDependency);
     }
 
+    std::vector<const Job *>
+    markIncrementalExternalInDepGraph(StringRef externalDependency,
+                                      const bool forRanges) {
+      return getFineGrainedDepGraph(forRanges)
+          .findIncrementalExternallyDependentUntracedJobs(externalDependency);
```

Comment:
Why only untraced jobs?

Reply:
Preserves the existing behavior of plain external dependencies.

Reply:
It's the old code's fault, but it would sure be nice if the reason were clearer.

Reply:
I think it was an optimization.